packages/next/client/router.ts

```diff
@@ -133,6 +134,7 @@ export function useRouter(): NextRouter {
 // This should **not** use inside the server.
 export const createRouter = (...args: RouterArgs): Router => {
   singletonRouter.router = new Router(...args)
+  singletonRouter.router.isReady = true
```

Comment:
The router is not ready when `createRouter` is called. Please add a test to use an auto exported dynamic route and you'll see what I mean!

[This is the criteria](https://github.com/vercel/next.js/blob/b0d2a80a18645b5b7278c8e72829e76063b91945/packages/next/client/index.tsx#L130-L134) if the router is already ready when `createRouter` is called.

Otherwise, you need to wait for [this re-render](https://github.com/vercel/next.js/blob/b0d2a80a18645b5b7278c8e72829e76063b91945/packages/next/client/index.tsx#L137-L159) to complete.

Reply:
Oh yeah, you're totally right!  I definitely realized my tests were bad on this not long after I made the latest change, but I've been super busy lately so I haven't updated this PR.

I will spend some time this weekend and correct the tests then use those links you provided to place this in the correct place.  As it is now, this is definitely not functioning like it should.

Thanks for the pointers!  They're super helpful.